### PR TITLE
Remove withRouter and replace history.push with goTo

### LINKED ISSUE #338: Use globally `goTo` function for navigation through pages
- [ ] Remove all unnecessary `withRouter` use and `props.history` calls
- [ ] Replace the `history.push` with the `goTo` function
- [ ] Make `goTo` function to be working both as an event handler and explicitely

### PR DESCRIPTION
**Pull request by vlad-bm**
(Closes #338 #317)

### CODE REVIEW

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentItems/DocumentItems.js [hunk 1]
```diff
@@ -21,12 +22,11 @@ export default class DocumentItems extends Component {
     this.fetchDocumentItems(this.props.document.document_pid);
   }
 
-  showDetailsHandler = itemPid =>
-    this.props.history.push(this.showDetailsUrl(itemPid));
+  showDetailsHandler = itemPid => goTo(this.showDetailsUrl(itemPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentItems/DocumentItems.js]: We can skip that declaration, and use directly `goToHandler` further down where it is used.

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentPendingLoans/DocumentPendingLoans.js [hunk 2]
```diff
@@ -22,13 +23,12 @@ export default class DocumentPendingLoans extends Component {
     this.fetchPendingLoans(document_pid);
   }
 
-  showDetailsHandler = loanPid =>
-    this.props.history.push(this.showDetailsUrl(loanPid));
+  showDetailsHandler = loanPid => goTo(this.showDetailsUrl(loanPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentPendingLoans/DocumentPendingLoans.js]: Same for this one

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/IdleLoansList/IdleLoansList.js [hunk 3]
```diff
@@ -21,12 +22,11 @@ export default class IdleLoansList extends Component {
     this.fetchIdlePendingLoans();
   }
 
-  showDetailsHandler = loanPid =>
-    this.props.history.push(this.showDetailsUrl(loanPid));
+  showDetailsHandler = loanPid => goTo(this.showDetailsUrl(loanPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/IdleLoansList/IdleLoansList.js]: Also this one

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/OverbookedDocumentsList/OverbookedDocumentsList.js [hunk 4]
```diff
@@ -20,12 +21,11 @@ export default class OverbookedDocumentsList extends Component {
     this.fetchOverbookedDocuments();
   }
 
-  showDetailsHandler = documentPid =>
-    this.props.history.push(this.showDetailsUrl(documentPid));
+  showDetailsHandler = documentPid => goTo(this.showDetailsUrl(documentPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/OverbookedDocumentsList/OverbookedDocumentsList.js]: And this one

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/OverdueLoansList/OverdueLoansList.js [hunk 5]
```diff
@@ -20,12 +21,11 @@ export default class OverdueLoansList extends Component {
     this.fetchOverdueLoans();
   }
 
-  showDetailsHandler = loanPid =>
-    this.props.history.push(this.showDetailsUrl(loanPid));
+  showDetailsHandler = loanPid => goTo(this.showDetailsUrl(loanPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/OverdueLoansList/OverdueLoansList.js]: And this one too 💃

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/RenewedLoansList/RenewedLoansList.js [hunk 6]
```diff
@@ -20,12 +21,11 @@ export default class RenewedLoansList extends Component {
...
   }
 
-  showDetailsHandler = loanPid =>
-    this.props.history.push(this.showDetailsUrl(loanPid));
+  showDetailsHandler = loanPid => goTo(this.showDetailsUrl(loanPid));
 
   seeAllButton = () => {
     const click = () => {
-      this.props.history.push(this.seeAllUrl(listQuery));
+      goTo(this.seeAllUrl(listQuery));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Home/components/lists/RenewedLoansList/RenewedLoansList.js]: We can skip the click function here

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemPastLoans/ItemPastLoans.js [hunk 7]
```diff
@@ -21,13 +22,12 @@ export default class ItemPastLoans extends Component {
     this.fetchPastLoans(item_pid);
   }
 
-  showDetailsHandler = loanPid =>
-    this.props.history.push(this.showDetailsUrl(loanPid));
+  showDetailsHandler = loanPid => goTo(this.showDetailsUrl(loanPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemPastLoans/ItemPastLoans.js]: We know what to do!

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LoanDetails/components/AvailableItems/AvailableItems.js [hunk 8]
```diff
@@ -22,13 +23,12 @@ export default class AvailableItems extends Component {
     this.fetchAvailableItems(this.props.loan.metadata.document_pid);
   }
 
-  showDetailsHandler = itemPid =>
-    this.props.history.push(this.showDetailsUrl(itemPid));
+  showDetailsHandler = itemPid => goTo(this.showDetailsUrl(itemPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/LoanDetails/components/AvailableItems/AvailableItems.js]: It looks it repeats itself

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronsSearch/PatronsSearch.js [hunk 9]
```diff
@@ -44,7 +45,7 @@ export class PatronsSearch extends Component {
           results={results}
           viewDetailsClickHandler={patronPid => {
             const path = BackOfficeRoutes.patronDetailsFor(patronPid);
-            this.props.history.push(path);
+            goTo(path);
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronsSearch/PatronsSearch.js]: less is more `goTo(BackOfficeRoutes.patronDetailsFor(patronPid));`

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/SeriesDetails/components/SeriesDocuments/SeriesDocuments.js [hunk 10]
```diff
@@ -21,7 +21,7 @@ export default class SeriesDocuments extends Component {
     this.fetchSeriesDocuments(this.props.series.series_pid);
   }
 
-  showDetailsHandler = documentPid => goTo(this.showDetailsUrl(documentPid))();
+  showDetailsHandler = documentPid => goTo(this.showDetailsUrl(documentPid));
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/SeriesDetails/components/SeriesDocuments/SeriesDocuments.js]: 🔢

File: invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/DocumentsSearch/DocumentsSearch.js [hunk 11]
```diff
@@ -87,7 +88,7 @@ export class DocumentsSearch extends Component {
           results={results}
           viewDetailsClickHandler={documentPid => {
             const path = FrontSiteRoutes.documentDetailsFor(documentPid);
-            this.props.history.push(path);
+            goTo(path);
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/DocumentsSearch/DocumentsSearch.js]: skip the `path` variable declaration

File: invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/components/Header/Header.js [hunk 12]
```diff
@@ -29,7 +30,7 @@ export default class Header extends Component {
           <Dropdown.Divider />
           <Dropdown.Item
             onClick={() => {
-              this.props.history.push(BackOfficeRoutes.home);
+              goTo(BackOfficeRoutes.home);
```
Comment (topless) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/components/Header/Header.js]: Couldn't we directly call `goToHandler` here? and skip the `() => ...`?

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/SeriesDetails/components/SeriesDocuments/SeriesDocuments.js (hunk 10 above)
Comment (vlad-bm) [reply invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/SeriesDetails/components/SeriesDocuments/SeriesDocuments.js]: I don't know this one :)

File: invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/__tests__/ResultsTable.js [hunk 13]
```diff
@@ -181,23 +171,16 @@ describe('ResultsTable tests', () => {
...
         showMaxRows={1}
-        history={historyFn}
         seeAllComponent={buttonObj()}
         rowActionClickHandler={() => {}}
       />
     );
 
     const button = component.find('TableFooter').find('button');
     button.simulate('click');
-    expect(mockedHistoryPush).toHaveBeenCalled();
```
Comment (equadon) [inline invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/__tests__/ResultsTable.js]: Shouldn't we still keep these asserts to make sure the page actually changed? This same applies to the other test files too.

**Review by equadon — DISMISSED** [review]

**Review by topless — DISMISSED** [review]

**Pull request merged**